database/src/pool/sqlite.rs

```diff
@@ -173,6 +173,16 @@ static MIGRATIONS: &[&str] = &[
         PRIMARY KEY(aid, cid, crate, profile, cache)
     );
     "#,
+    // Add not null constraint to benchmark name.
+    r#"
+    create table benchmark_new(
+        name text primary key not null,
+        stabilized bool not null
+    );
+    insert into benchmark_new select * from benchmark where name is not null;
+    drop table benchmark;
+    alter table benchmark_new rename to benchmark;
```

Comment:
Hm, this seems like it does not quite follow the rules laid out in the docs (https://www.sqlite.org/lang_altertable.html)

In particular, I'd expect this to fail on the drop table if it's run on an existing (not empty) db, since there are foreign key constraints onto the benchmark table's existence. (Maybe those get deleted when the table is dropped, but that is also bad).

That said those docs at a skim are a little unclear -- I am not sure if step 8 is necessary; it's not called out in the box, but it is present in the instructions...

Reply:
I'll look into fixing this if it's not too difficult. I don't think this migration is particularly important, but the fix could be usable for future "complex" migrations, too.

Reply:
Yeah. I'm by no means an expert, so it's probably worth checking that this isn't already good enough - it might be.

Reply:
Fixed. Thanks for catching this. By the way, when foreign key checking is enabled, `DROP TABLE` does in fact trigger any `ON DELETE` actions in other tables referencing the dropped rows (cascade delete in this case).